<a href="https://colab.research.google.com/github/gauravrs195/Capstone-1-Play-Store-App-Review-Analysis/blob/main/Copy_of_Play_Store_App_Review_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

##Exploring Play Store Data

In [2]:
# import libraries
import numpy as np
import pandas as pd
import matplotlib
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
from datetime import datetime

#**Import Playstore Data**

In [3]:
# mounting drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
#load data
file_path = '/content/drive/MyDrive/Colab Notebooks/AlmaBetter/Modules/Python for data science/EDA Project/Play Store Data.csv'
psd_df= pd.read_csv(file_path)

In [5]:
#First 5 rows of the play store dataframe
psd_df.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [6]:
# getting the last five rows
psd_df.tail()

App             Category  \
10836                               Sya9a Maroc - FR               FAMILY   
10837               Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                         Parkinson Exercices FR              MEDICAL   
10839                  The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840  iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating             Genres      Last Updated         Current Ver  \
10836       Everyone          Education     July 25, 2017                1.48   
10837       Everyone          Education      July 6, 2018                 1.0   
10838       Everyone            Medical  January 20, 2017                 1.0   
10839     Mature 17+  Books & Reference  January 19, 2015  Varies with device   
10840       Everyone          Lifestyle     July 25, 2018  Varies with device   

              Android Ver  
10836          4.1 and up  
10837          4.1 and up  
10838          2.2 and up  
10839  Varies with device  
10840  Varies with device

In [7]:
psd_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [8]:
# Print the Shape of the Two DataFrames
psd_df.shape

(10841, 13)

In [9]:
# column names
psd_df.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

In [10]:
psd_df.describe()

Rating
count  9367.000000
mean      4.193338
std       0.537431
min       1.000000
25%       4.000000
50%       4.300000
75%       4.500000
max      19.000000

##Cleaning of the data

In [11]:
# Removing the null value and Duplicate present in the data set

In [12]:
# this function give type, non null value, null value, unique value and non null value %
def playstoreinfo():
  temp = pd.DataFrame(index = psd_df.columns)
  temp['Data Type'] = psd_df.dtypes
  temp['non null value'] = psd_df.count()
  temp['null value'] = psd_df.isnull().sum()
  temp['unique values'] = psd_df.nunique()
  temp['non null value percentage'] = psd_df.isnull().mean()
  return temp
playstoreinfo()

Data Type  non null value  null value  unique values  \
App               object           10841           0           9660   
Category          object           10841           0             34   
Rating           float64            9367        1474             40   
Reviews           object           10841           0           6002   
Size              object           10841           0            462   
Installs          object           10841           0             22   
Type              object           10840           1              3   
Price             object           10841           0             93   
Content Rating    object           10840           1              6   
Genres            object           10841           0            120   
Last Updated      object           10841           0           1378   
Current Ver       object           10833           8           2832   
Android Ver       object           10838           3             33   

                non null value percentage  
App                              0.000000  
Category                         0.000000  
Rating                           0.135965  
Reviews                          0.000000  
Size                             0.000000  
Installs                         0.000000  
Type                             0.000092  
Price                            0.000000  
Content Rating                   0.000092  
Genres                           0.000000  
Last Updated                     0.000000  
Current Ver                      0.000738  
Android Ver                      0.000277

**Findings**

The number of null values are:
1. **Rating** has 1474 null values which contributes **13.60%** of the data.
2. **Type** has 1 null value which contributes **0.01%** of the data.
3. **Content_Rating** has 1 null value which contributes **0.01%** of the data.
4. **Current_Ver** has 8 null values which contributes **0.07%** of the data.
5. **Android_Ver** has 3 null values which contributes **0.03%** of the data.

#Android Ver - There are 3 null Value

In [13]:
# Null value in Android Ver column
psd_df[psd_df['Android Ver'].isnull()]

App         Category  Rating  \
4453                    [substratum] Vacuum: P  PERSONALIZATION     4.4   
4490                      Pi Dark [substratum]  PERSONALIZATION     4.5   
10472  Life Made WI-Fi Touchscreen Photo Frame              1.9    19.0   

      Reviews    Size Installs  Type     Price Content Rating  \
4453      230     11M   1,000+  Paid     $1.49       Everyone   
4490      189    2.1M  10,000+  Free         0       Everyone   
10472    3.0M  1,000+     Free     0  Everyone            NaN   

                  Genres    Last Updated Current Ver Android Ver  
4453     Personalization   July 20, 2018         4.4         NaN  
4490     Personalization  March 27, 2018         1.1         NaN  
10472  February 11, 2018          1.0.19  4.0 and up         NaN

In [14]:
# delete Null value in Android Ver column
psd_df = psd_df[psd_df['Android Ver'].notna()]

# New shape of dataframe after deleting null value in android ver columns
psd_df.shape

(10838, 13)

In [15]:
# checking for NaN values in diiferent columns
psd_df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       0
Genres               0
Last Updated         0
Current Ver          8
Android Ver          0
dtype: int64

#Current Ver - 8 Null value in this columns

In [16]:
# Null value in Current Ver
psd_df[psd_df['Current Ver'].isnull()]

App             Category  Rating Reviews  \
15     Learn To Draw Kawaii Characters       ART_AND_DESIGN     3.2      55   
1553              Market Update Helper   LIBRARIES_AND_DEMO     4.1   20145   
6322            Virtual DJ Sound Mixer                TOOLS     4.2    4010   
6803                         BT Master               FAMILY     NaN       0   
7333                       Dots puzzle               FAMILY     4.0     179   
7407                   Calculate My IQ               FAMILY     NaN      44   
7730                            UFO-CQ                TOOLS     NaN       1   
10342                   La Fe de Jesus  BOOKS_AND_REFERENCE     NaN       8   

       Size    Installs  Type  Price Content Rating             Genres  \
15     2.7M      5,000+  Free      0       Everyone       Art & Design   
1553    11k  1,000,000+  Free      0       Everyone   Libraries & Demo   
6322   8.7M    500,000+  Free      0       Everyone              Tools   
6803   222k        100+  Free      0       Everyone          Education   
7333    14M     50,000+  Paid  $0.99       Everyone             Puzzle   
7407   7.2M     10,000+  Free      0       Everyone      Entertainment   
7730   237k         10+  Paid  $0.99       Everyone              Tools   
10342  658k      1,000+  Free      0       Everyone  Books & Reference   

            Last Updated Current Ver Android Ver  
15          June 6, 2018         NaN  4.2 and up  
1553   February 12, 2013         NaN  1.5 and up  
6322        May 10, 2017         NaN  4.0 and up  
6803    November 6, 2016         NaN  1.6 and up  
7333      April 18, 2018         NaN  4.0 and up  
7407       April 3, 2017         NaN  2.3 and up  
7730        July 4, 2016         NaN  2.0 and up  
10342   January 31, 2017         NaN  3.0 and up

In [17]:
# delete NaN value in Current Ver column
psd_df = psd_df[psd_df['Current Ver'].notna()]

# new shape of dataframe
psd_df.shape


(10830, 13)

In [18]:
# checking for NaN values in diiferent columns
psd_df.isnull().sum()

App                  0
Category             0
Rating            1470
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       0
Genres               0
Last Updated         0
Current Ver          0
Android Ver          0
dtype: int64

#Type - There is 1 NaN value in this columns

In [19]:
psd_df[psd_df['Type'].isnull()]

App Category  Rating Reviews                Size  \
9148  Command & Conquer: Rivals   FAMILY     NaN       0  Varies with device   

     Installs Type Price Content Rating    Genres   Last Updated  \
9148        0  NaN     0   Everyone 10+  Strategy  June 28, 2018   

             Current Ver         Android Ver  
9148  Varies with device  Varies with device

The `Type `column have two type of entities, i.e. `Free` and `Paid`. If the type is paid then the price will be printed in `Price` column, else, it will show as '0'. In this case, the price of app is printed as '0', which means the app is of type-free. Hence we can replace this NaN value with Free.

In [20]:
# Replacing the NaN value in 'Type' column corresponding to row index 9148 with 'Free'
psd_df.loc[9148,'Type'] = 'Free'

In [21]:
# checking for null values in diiferent columns
psd_df.isnull().sum()

App                  0
Category             0
Rating            1470
Reviews              0
Size                 0
Installs             0
Type                 0
Price                0
Content Rating       0
Genres               0
Last Updated         0
Current Ver          0
Android Ver          0
dtype: int64